pyproject.toml

```diff
@@ -0,0 +1,223 @@
+    "stevedore>=1.20.0",
+    "chardet>=2.3.0",
+    "appdirs>=1.3",
+    "rarfile>=2.7",
+    "srt",
+    "pytz>=2012c",
+]
+
+# extras
+# https://peps.python.org/pep-0621/#dependencies-optional-dependencies
+[project.optional-dependencies]
+docs = [
+    "sphinx",
+    "sphinx_rtd_theme",
+    "sphinxcontrib-programoutput",
+]
+test = [
+    "lxml",
+    "pytest>=6.0",
+    "pytest-cov",
+    "pytest-flakes",
+    "sympy",
+    "vcrpy>=1.6.1",
+]
+dev = [
+    "mypy",
+    "ruff",
+    "typos",
+    "validate-pyproject",
+    "tox",
+]
+
+[project.urls]
+homepage = "https://github.com/Diaoul/subliminal"
+repository = "https://github.com/Diaoul/subliminal"
+documentation = "https://subliminal.readthedocs.org"
+
+[project.scripts]
+subliminal = "subliminal.cli:subliminal"
+
+[project.entry-points."subliminal.providers"]
+addic7ed = "subliminal.providers.addic7ed:Addic7edProvider"
+argenteam = "subliminal.providers.argenteam:ArgenteamProvider"
+gestdown = "subliminal.providers.gestdown:GestdownProvider"
+napiprojekt = "subliminal.providers.napiprojekt:NapiProjektProvider"
+opensubtitles = "subliminal.providers.opensubtitles:OpenSubtitlesProvider"
+opensubtitlesvip = "subliminal.providers.opensubtitles:OpenSubtitlesVipProvider"
+podnapisi = "subliminal.providers.podnapisi:PodnapisiProvider"
+shooter = "subliminal.providers.shooter:ShooterProvider"
+thesubdb = "subliminal.providers.thesubdb:TheSubDBProvider"
+tvsubtitles = "subliminal.providers.tvsubtitles:TVsubtitlesProvider"
+
+[project.entry-points."subliminal.refiners"]
+hash = "subliminal.refiners.hash:refine"
+metadata = "subliminal.refiners.metadata:refine"
+omdb = "subliminal.refiners.omdb:refine"
+tvdb = "subliminal.refiners.tvdb:refine"
+
+[project.entry-points."babelfish.language_converters"]
+addic7ed = "subliminal.converters.addic7ed:Addic7edConverter"
+shooter = "subliminal.converters.shooter:ShooterConverter"
+thesubdb = "subliminal.converters.thesubdb:TheSubDBConverter"
+tvsubtitles = "subliminal.converters.tvsubtitles:TVsubtitlesConverter"
+
+[tool.setuptools]
+py-modules = ["subliminal"]
+include-package-data = true
+
+[tool.setuptools.package-data]
+subliminal = [
+    "py.typed",
+]
+
+[tool.setuptools.packages.find]
+namespaces = false
+where = ["."]
+
+[tool.setuptools.dynamic]
+version = {attr = "subliminal.__version__"}
+readme = {file = ["README.rst", "HISTORY.rst"]}
+
+
+# https://docs.astral.sh/ruff/
+[tool.ruff]
+line-length = 120
+src = ["subliminal", "tests"]
+
+[tool.ruff.lint]
+pydocstyle = { convention = "pep257" }
+select = [
+    "E",    # style errors
+    "F",    # flakes
+    "W",    # warnings
+    "D",    # pydocstyle
+    "D417", # Missing argument descriptions in Docstrings
+    "I",    # isort
+    "UP",   # pyupgrade
+    "S",    # bandit
+    "C4",   # flake8-comprehensions
+    "B",    # flake8-bugbear
+    "A001", # flake8-builtins
+    "TCH",  # flake8-typecheck
+    "TID",  # flake8-tidy-imports
+    "RUF",  # ruff-specific rules
+]
```

Comment:
Is this necessary or can we go with the default?

Reply:
default ruff config doesn't include much, unfortunately

Reply:
Ah bummer and very counterintuitive :disappointed:

Reply:
One alternative (that I use in personal project) is to put `select = ["ALL"]` and ignore specific rules.
Like that, if a new version of ruff adds new rules, they are automatically included.
But then we would need to pin ruff version to avoid adding rules automatically...

Reply:
i don't think we'd want to do `select = ["ALL"]` ( e.g. I use my system `ruff` install which could be updated separately from subliminal, so that would make it fail for just me but not others )

Reply:
I agree